analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -147,4 +147,23 @@ public override void ExecutionCompleted()
             ReportIssue(operation, operation.Syntax.ToString());
         }
     }
+
+    private static ProgramState SetConstraintForObjectCreation(ProgramState state, IObjectCreationOperationWrapper objectCreation) =>
+        FirstArgumentAsCollection(objectCreation) switch
+        {
+            { } firstArgument when ConstraintFromArgument(state, firstArgument) is { } constraint => state.SetOperationConstraint(objectCreation, constraint),
+            { } => state,
+            _ => state.SetOperationConstraint(objectCreation, CollectionConstraint.Empty)
+        };
+
+    private static IArgumentOperationWrapper? FirstArgumentAsCollection(IObjectCreationOperationWrapper objectCreation) =>
+        objectCreation.Arguments.Select(x => x.AsArgument()).SingleOrDefault(x => x?.Parameter.Ordinal == 0) is { } firstArgument
```

Comment:
This requires the line to be re-written like this, which is way harder to read:
`        objectCreation.Arguments.Select(x => x.ToArgument() as IArgumentOperationWrapper?).SingleOrDefault(x => x?.Parameter.Ordinal == 0) is { } firstArgument`
Or do you see an alternative?

Reply:
Why would you do `as IArgumentOperationWrapper`? The `ToArgument` already returns `IArgumentOperationWrapper`

Reply:
And it can be further simplified with the idea that we don't need to cast it, returning `IOperation` is enough

Reply:
It does not work without the `as IArgumentOperationWrapper?` because `SingleOrDefault` will not return `null` for a struct, and there is no easy way to check for `default`.
Omitting the check on `Ordinal` simplifies it enough to work around the problem. I have checked all covered collections, and no constructor takes a different kind of collection as a parameter. So it should be safe.